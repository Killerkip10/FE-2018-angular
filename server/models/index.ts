export {User} from './user';
export {Login} from './login';
export {ShortUser} from './short-user';
