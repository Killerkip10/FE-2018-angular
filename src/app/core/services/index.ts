export * from './auth.service';
export * from './profile.service';
export * from './token.service';
export * from './search-user.service';
