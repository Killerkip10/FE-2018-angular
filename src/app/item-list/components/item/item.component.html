<div class="item" *ngIf="node?.open">
  <span (click)="addItems($event)">{{node?.value}}</span>

  <div *ngIf="node?.children">
    <app-item
      *ngFor="let value of node?.children"
      [node]="value"
      (select)="onSelect($event)"></app-item>
  </div>
</div>
