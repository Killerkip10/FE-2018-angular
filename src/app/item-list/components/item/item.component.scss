.item{
  margin-left: 20px;
  span{
    cursor: pointer;
    &:hover{
      color: #3f51b5;
      background-color: rgb(245, 245, 245);
    }
  }
}
