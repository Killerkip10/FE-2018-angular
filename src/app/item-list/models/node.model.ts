export interface Node {
  value: string;
  open: boolean;
  children: Node[];
}
