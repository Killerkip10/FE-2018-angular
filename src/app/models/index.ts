export * from '../../../server/models/index';
