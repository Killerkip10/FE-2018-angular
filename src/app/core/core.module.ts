import {NgModule} from '@angular/core';
import {BrowserModule} from '@angular/platform-browser';
import {TranslateModule} from '@ngx-translate/core';
import {HttpClientModule} from '@angular/common/http';
import {RouterModule} from '@angular/router';

import {BrowserAnimationsModule} from '@angular/platform-browser/animations';
import {AuthGuard} from './guards';
import {AuthService} from './services';
import {
  MatButtonModule,
  MatToolbarModule,
  MatSelectModule,
  MatIconModule
} from '@angular/material';
import {
  HeaderComponent,
  FooterComponent
} from './components';

@NgModule({
  imports: [
    BrowserModule,
    HttpClientModule,
    TranslateModule,
    BrowserAnimationsModule,

    RouterModule,

    MatIconModule,
    MatButtonModule,
    MatSelectModule,
    MatToolbarModule
  ],
  declarations: [
    HeaderComponent,
    FooterComponent
  ],
  providers: [
    AuthService,

    AuthGuard
  ],
  exports: [
    HeaderComponent,
    FooterComponent
  ]
})
export class CoreModule {}
