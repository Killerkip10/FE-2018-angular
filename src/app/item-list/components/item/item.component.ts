import {ChangeDetectionStrategy, Component, EventEmitter, Input, Output} from '@angular/core';

import {Node} from '../../models';

@Component({
  selector: 'app-item',
  templateUrl: './item.component.html',
  styleUrls: ['./item.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class ItemComponent {
  @Input() node: Node;
  @Output() select = new EventEmitter<Node>();

  constructor() {}

  public addItems(): void {
    if (this.node.children) {
      this.node.children = null;
    } else {
      this.select.emit(this.node);
    }
  }
  public onSelect(node): void {
    this.select.emit(node);
  }
}
