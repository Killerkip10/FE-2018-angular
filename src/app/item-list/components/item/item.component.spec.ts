import {async, ComponentFixture, TestBed} from '@angular/core/testing';
import {ChangeDetectionStrategy, NO_ERRORS_SCHEMA} from '@angular/core';
import {CommonModule} from '@angular/common';

import {ItemComponent} from './item.component';
import {Node} from '../../models';
import {CoreModule} from '../../../core/core.module';

describe('ItemComponent', () => {
  let component: ItemComponent;
  let fixture: ComponentFixture<ItemComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        CoreModule,
        CommonModule
      ],
      declarations: [
        ItemComponent
      ],
      schemas: [
        NO_ERRORS_SCHEMA
      ]
    })
      .overrideComponent(ItemComponent, {set: {changeDetection: ChangeDetectionStrategy.Default}})
      .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ItemComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('snapshot', () => {
    component.node = {} as Node;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('snapshot', () => {
    component.node = {
      value: 'item 1',
      open: true,
      children: null
    } as Node;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('snapshot', () => {
    component.node = {
      value: 'item 1',
      open: false,
      children: null
    } as Node;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('snapshot', () => {
    component.node = {
      value: 'item 1',
      open: true,
      children: [
        {
          value: 'item 1',
          open: true,
          children: null
        },
        {
          value: 'item 2',
          open: true,
          children: null
        },
      ]
    } as Node;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
