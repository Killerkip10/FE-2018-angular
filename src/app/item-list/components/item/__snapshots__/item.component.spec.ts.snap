// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemComponent snapshot 1`] = `
<app-item
  select={[Function EventEmitter]}
>
  
</app-item>
`;

exports[`ItemComponent snapshot 2`] = `
<app-item
  node={[Function Object]}
  select={[Function EventEmitter]}
>
  
</app-item>
`;

exports[`ItemComponent snapshot 3`] = `
<app-item
  node={[Function Object]}
  select={[Function EventEmitter]}
>
  <div
    class="item"
  >
    <span>
      item 1
    </span>
    
  </div>
</app-item>
`;

exports[`ItemComponent snapshot 4`] = `
<app-item
  node={[Function Object]}
  select={[Function EventEmitter]}
>
  
</app-item>
`;

exports[`ItemComponent snapshot 5`] = `
<app-item
  node={[Function Object]}
  select={[Function EventEmitter]}
>
  <div
    class="item"
  >
    <span>
      item 1
    </span>
    
    <div>
      
      <app-item
        ng-reflect-node="[object Object]"
      >
        
        <div
          class="item"
        >
          <span>
            item 1
          </span>
          
        </div>
      </app-item>
      <app-item
        ng-reflect-node="[object Object]"
      >
        
        <div
          class="item"
        >
          <span>
            item 2
          </span>
          
        </div>
      </app-item>
    </div>
  </div>
</app-item>
`;
